CCM-7747: Remove DOB

simonlabarere opened the pull request:
## Summary

Remove DOB validation


## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [ ] Brief description of work completed, and any technical decisions made as part of the PR
* [ ] PR link added as a comment to the relevant JIRA ticket
* [ ] PR link shared on Slack and/or Teams
* [ ] 2 reviews received
* [ ] Tester approval

aaron-downing commented on tests/development/message_batches/create_message_batches/test_success.py:
```diff
@@ -106,45 +106,8 @@ def test_201_message_batch_valid_contact_details(
-            "Content-Type": constants.DEFAULT_CONTENT_TYPE,
-        },
-        json=data,
+    Assertions.assert_201_response(
```
I'm guessing this needs the latest main pulling in?

simonlabarere replied:
Yep, done!

adadigital commented:
I found references to dateOfBirth still in `generate_valid_create_message_batch_body` in `tests/lib/generators.py` on lines 29 and 40. Does anything depend on these?

tdroza-nhs commented on proxies/shared/resources/jsc/helpers/validationChecks.js:
```diff
@@ -106,21 +106,6 @@ const validateArray = (errors, fieldValue, fieldPointer, minElements) => {
-        && (
-            fieldValue === null
-            || typeof fieldValue !== "string"
-            || !dobRegex.test(fieldValue)
```
Looks like you can also remove line 4 in this file?
```
const dobRegex = /^\d{4}-\d{2}-\d{2}$/i
```

simonlabarere replied:
Nice catch! Done.

simonlabarere commented:
> sandbox/messages/2WL4g2f7GQ2F9HZmgonIFiXAPku.json
It looks like this was only referenced in the sandbox postman collection so I removed it from there as well.

> I found references to dateOfBirth still in `generate_valid_create_message_batch_body` in `tests/lib/generators.py` on lines 29 and 40. Does anything depend on these?
Those are now removed as well.
